nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -376,7 +682,21 @@ in
         zfsSupport = true;
       };
 
-      environment.etc."zfs/zed.d".source = "${packages.zfsUser}/etc/zfs/zed.d/";
+      
+      environment.etc."zfs/zed.d/all-syslog.sh".source = "${packages.zfsUser}/etc/zfs/zed.d/all-syslog.sh";
+      environment.etc."zfs/zed.d/data-notify.sh".source = "${packages.zfsUser}/etc/zfs/zed.d/data-notify.sh";
+      environment.etc."zfs/zed.d/pool_import-led.sh".source = "${packages.zfsUser}/etc/zfs/zed.d/pool_import-led.sh";
+      environment.etc."zfs/zed.d/resilver_finish-notify.sh".source = "${packages.zfsUser}/etc/zfs/zed.d/resilver_finish-notify.sh";
+      environment.etc."zfs/zed.d/resilver_finish-start-scrub.sh".source = "${packages.zfsUser}/etc/zfs/zed.d/resilver_finish-start-scrub.sh";
+      environment.etc."zfs/zed.d/scrub_finish-notify.sh".source = "${packages.zfsUser}/etc/zfs/zed.d/scrub_finish-notify.sh";
+      environment.etc."zfs/zed.d/statechange-led.sh".source = "${packages.zfsUser}/etc/zfs/zed.d/statechange-led.sh";
+      environment.etc."zfs/zed.d/statechange-notify.sh".source = "${packages.zfsUser}/etc/zfs/zed.d/statechange-notify.sh";
+      environment.etc."zfs/zed.d/vdev_attach-led.sh".source = "${packages.zfsUser}/etc/zfs/zed.d/vdev_attach-led.sh";
+      environment.etc."zfs/zed.d/vdev_clear-led.sh".source = "${packages.zfsUser}/etc/zfs/zed.d/vdev_clear-led.sh";
+      environment.etc."zfs/zed.d/zed-functions.sh".source = "${packages.zfsUser}/etc/zfs/zed.d/zed-functions.sh";
```

Comment:
You can abstract this:
```nix
{
  environment.etc = genAttrs (map (file: "zfs/zed.d/${file}") [
    "all-syslog.sh"
    "data-notify.sh"
    # ... 
  ]) (file: { source = "${packages.zfsUser}/etc/${file}"; });
}
```

Reply:
This doesn't seem resolved

Reply:
why would this abstraction be better except making it more difficult to comprehend/read?

Reply:
Prevents some typos, is simpler, DRY, looks better

Reply:
How would it prevent typos?
It's not really simpler IMHO.
Not sure what DRY is?
Looks better -  beauty is in the eye of the beholder.

Mor interesting would be to loop through the .sh files in a generic way than specifying each on of those but I have no idea how to do that in a module file.

Reply:
It can prevent typos because you don't have to repeat the name of the files (which is what DRY stands for, Don't Repeat Yourself)